src/mon/OSDMonitor.cc

```diff
@@ -7925,10 +7925,22 @@ int OSDMonitor::prepare_command_pool_set(const cmdmap_t& cmdmap,
       ss << "error parsing integer value '" << val << "': " << interr;
       return -EINVAL;
     }
+    if (!g_conf()->mon_allow_pool_size_one) {
+      ss << "configuring pool without replicas is disabled by default; you must first set the mon_allow_pool_size_one config option to true before you can set pool size to 1";
+      return -EPERM;
+    }
     if (n <= 0 || n > 10) {
       ss << "pool size must be between 1 and 10";
       return -EINVAL;
     }
+    bool sure = false;
+    cmd_getval(cmdmap, "yes_i_really_mean_it", sure);
+    if (n == 1) {
+    if (!sure) {
```

Comment:
https://github.com/ceph/ceph/blob/db5b1d6c29a2ca4c3784397932adb4f8d24ba8ef/src/mon/OSDMonitor.cc#L7939 indentation still needs to be fixed